src/Login/Login.js

```diff
@@ -23,15 +23,26 @@ class Login extends Component {
     const loginInfo = {
       ...this.state
     }
-    this.props.handleSubmit(loginInfo)
-    this.props.history.push('/')
+    if(this.formFulfilled()) {
```

Comment:
Great error handling here! I like that you're disabling the submit if all fields are not logged out & also displaying a message to the user.